test/CMakeLists.txt

```diff
@@ -80,13 +80,23 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU"
         if(NOT CMAKE_CXX_FLAGS MATCHES "-march")
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${TARGET_ARCH}")
         endif()
-        CHECK_CXX_COMPILER_FLAG("-std=c++11" HAS_CPP11_FLAG)
-        if (ENABLE_XTL_COMPLEX)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fPIC -Wunused-parameter -Wextra -Wreorder -std=c++14")
-        elseif (HAS_CPP11_FLAG)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fPIC -Wunused-parameter -Wextra -Wreorder -std=c++11")
-        else()
-            message(FATAL_ERROR "Unsupported compiler -- xsimd requires C++11 support!")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fPIC -Wunused-parameter -Wextra -Wreorder")
+        # Users may override the c++ standard:
+        if(NOT DEFINED CMAKE_CXX_STANDARD OR "${CMAKE_CXX_STANDARD}" STREQUAL "")
+            if (ENABLE_XTL_COMPLEX)
+                CHECK_CXX_COMPILER_FLAG("-std=c++14" HAS_CPP14_FLAG)
+                if (NOT HAS_CPP11_FLAG)
+                    message(FATAL_ERROR "Unsupported compiler -- xsimd requires C++11 support when complex support is enabled")
```

Comment:
typo: that would be C++14 in the comment here. And it would be 'when xtl complex support is enabled'